### PR TITLE
ceph: correctly return if multus was applied

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

We must only return if multus got configured successfully.

Signed-off-by: Sébastien Han <seb@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/rook/rook/issues/6151

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/csi/spec.go [hunk 1]
```diff
@@ -523,6 +523,7 @@ func deleteCSIDriverResources(
 }
 
 func applyCephClusterNetworkConfig(objectMeta *metav1.ObjectMeta, rookclientset rookclient.Interface) (bool, error) {
+	var isMultusApplied bool
 	cephClusters, err := rookclientset.CephV1().CephClusters(objectMeta.Namespace).List(metav1.ListOptions{})
```
Comment (Madhu-1) [inline pkg/operator/ceph/csi/spec.go:527]: not related to this PR, but still a  question `objectMeta.Namespace` is the namespace of the operator as csi pods will start in rook operator namespace. will this check passes if we have ceph clusters in different namespaces?

**Review by Madhu-1 — APPROVED** [review]
LGTM

File: pkg/operator/ceph/csi/spec.go (hunk 1 above)
Comment (leseb) [reply pkg/operator/ceph/csi/spec.go:527]: @rohan47 is working on a fix for this.

Comment (leseb) [reply pkg/operator/ceph/csi/spec.go:527]: So if the CephCluster is not in the operator namespace this won't work.

File: pkg/operator/ceph/csi/spec.go [hunk 2]
```diff
@@ -533,10 +534,11 @@ func applyCephClusterNetworkConfig(objectMeta *metav1.ObjectMeta, rookclientset
 			if err != nil {
 				return false, errors.Wrapf(err, "failed to apply multus configuration to CephCluster %q", cephCluster.Name)
 			}
+			isMultusApplied = true
 		}
 	}
 
-	return true, nil
+	return isMultusApplied, nil
```
Comment (humblec) [inline pkg/operator/ceph/csi/spec.go:541]: @leseb do we need this bool ? or cant we just survive with 'error' return in caller for nil , or error ?  :)

Comment (leseb) [reply pkg/operator/ceph/csi/spec.go:541]: The bool checks whether multus is applied and error checks for any underlaying error so I think it's cleaner this way :)

**Pull request merged**